data/src/test/java/org/akvo/flow/data/repository/DataPointDataRepositoryTest.kt

```diff
@@ -19,150 +19,126 @@
-        ""
-    )
-
-    private lateinit var spyHttpException: HttpException
-
-    private lateinit var spyRestApi: RestApi
+    lateinit var repository: DataPointDataRepository
 
     @Before
     fun setUp() {
-        `when`(mapper.getImagesList(anyList())).thenReturn(emptyList())
-        spyRestApi = spy(RestApi(mockDeviceHelper, TestRestServiceFactory(), "", ""))
-        spyHttpException = spy(HttpException(retrofit2.Response.success("")))
-        `when`(mockDataSourceFactory.dataBaseDataSource).thenReturn(mockDatabaseDataSource)
-    }
-
-    @Test(expected = AssignmentRequiredException::class)
-    fun downloadDataPointsShouldReturnExpectedErrorWhenAssignmentMissing() = runBlockingTest {
-        doThrow(spyHttpException).`when`(spyRestApi).downloadDataPoints(anyLong())
-        doReturn(HttpURLConnection.HTTP_FORBIDDEN).`when`(spyHttpException).code()
-
-        val repository =
+        MockKAnnotations.init(this)
+        every { mockMapper.getImagesList(any()) } returns emptyList()
+        every { mockDataSourceFactory.dataBaseDataSource } returns mockDatabaseDataSource
+        every { mockDatabaseDataSource.getDataPointCursor(any()) } returns null
+        repository =
             DataPointDataRepository(
                 mockDataSourceFactory,
-                spyRestApi,
+                mockRestApi,
                 mockS3RestApi,
-                mapper,
+                mockMapper,
                 mediaHelper
             )
-
-        repository.downloadDataPoints(123L)
     }
 
-    @Test(expected = HttpException::class)
-    fun downloadDataPointsShouldReturnExpectedErrorWhenNotAssignmentMissing() = runBlockingTest {
-        doThrow(spyHttpException).`when`(spyRestApi).downloadDataPoints(anyLong())
-        doReturn(HttpURLConnection.HTTP_BAD_GATEWAY).`when`(spyHttpException).code()
-
-        val repository =
-            DataPointDataRepository(
-                mockDataSourceFactory,
-                spyRestApi,
-                mockS3RestApi,
-                mapper,
-                mediaHelper
+    @Test
+    fun downloadDataPointsShouldReturnExpectedErrorWhenAssignmentMissing() = runBlocking {
+        coEvery {
+            mockRestApi.downloadDataPoints(
+                any(),
+                any()
             )
-
-        repository.downloadDataPoints(123L)
+        } coAnswers { throw spyHttpException }
+        every { spyHttpException.code() } returns HttpURLConnection.HTTP_FORBIDDEN
+
+        try {
+            val result = repository.downloadDataPoints(123L)
+        } catch (e: Exception) {
+            assert(e is AssignmentRequiredException)
+        }
     }
 
-    @Test(expected = Exception::class)
-    fun downloadDataPointsShouldReturnAnyErrorWhenNotAssignmentMissing()  = runBlockingTest {
-        doThrow(Exception()).`when`(spyRestApi).downloadDataPoints(anyLong())
+    @Test
+    fun downloadDataPointsShouldReturnExpectedErrorWhenNotAssignmentMissing() = runBlocking {
```

Comment:
its not very clear what this one means.. does that mean the assignment actually exists? if so why the exception?

Reply:
any other error we just rethrow, I will rename to clarify the method